parcels/particleset/particlesetsoa.py

```diff
@@ -173,7 +180,43 @@ def ArrayClass_init(self, *args, **kwargs):
             self.repeatkwargs = kwargs
 
         ngrids = fieldset.gridset.size if fieldset is not None else 1
-        self._collection = ParticleCollectionSOA(_pclass, lon=lon, lat=lat, depth=depth, time=time, lonlatdepth_dtype=lonlatdepth_dtype, pid_orig=pid_orig, partitions=partitions, ngrid=ngrids, **kwargs)
+        interaction_xy = None
+        interaction_z = None
+        self._dirty_neighbor = True
+        self._neighbor_time = None
+
+        self._collection = ParticleCollectionSOA(
+            _pclass, lon=lon, lat=lat, depth=depth, time=time,
+            lonlatdepth_dtype=lonlatdepth_dtype, pid_orig=pid_orig,
+            partitions=partitions, ngrid=ngrids, **kwargs)
+        if interaction_distance is not None:
+            meshes = [g.mesh for g in fieldset.gridset.grids]
+            # Assert all grids have the same mesh type
+            assert np.all(np.array(meshes) == meshes[0])
+            mesh_type = meshes[0]
+            if mesh_type == "spherical":
+                if len(self) < 1000:
+                    interaction_class = BruteSphericalNeighborSearch
+                else:
+                    interaction_class = HashSphericalNeighborSearch
```

Comment:
Ah, now I understand the difference between brute-force and hash! 
Has this value of 1000 been optimised already? Otherwise add a note/comment that it's tbd whether 1000 indeed is a good cut-off?

Reply:
The answer if `1000` is a good cut-off can't be easily answered. That would depend a lot on the initial distribution of all the particles (e.g. sparse, cluttered, uniform), the chosen interaction distance etc. The value is quite arbitrary. @qubixes : is this also part of the initial performance experiments you ran ? Or could that be part of a later performance study to find a representable cut-of point- or range ?

Reply:
@CKehl I haven't perfectly optimized it, but it should be in the right range. It was indirectly part of the performance experiments (compared n=999 and n=1000).

Reply:
nice!